tests/thread/host/spinlock_test_host.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <openenclave/host.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <signal.h>
+#include <stdio.h>
+#include <cassert>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <thread>
+#include "thread_u.h"
+
+#define ERRNO_TEST_TIMES 10
+#define LARGE_ENOUGH_MEMORY_SIZE 1024 * 1024 * 1024
+#define NUM_SPINLOCK_TEST_THREADS 2
+
+int g_flag_withlock = 0;
+static uint32_t g_lock = 0;
+static int g_error_num = 0;
+
+void* errno_thread1(void* para)
+{
+    int i = 0;
+    int t_errno = 0;
+    void* buffer = NULL;
+    oe_enclave_t* enclave = (oe_enclave_t*)para;
+
+    for (i = 0; i < ERRNO_TEST_TIMES; i++)
+    {
+        if (g_flag_withlock)
+        {
+            enc_spin_lock(enclave, &g_lock);
+        }
+
+        enc_malloc(enclave, &buffer, LARGE_ENOUGH_MEMORY_SIZE, &t_errno);
+
+        if (ENOMEM != t_errno)
+        {
+            printf(
+                "[errno_thread1]Error check errno, expected %d, got %d.\n",
+                ENOMEM,
+                t_errno);
+            g_error_num |= 0x8;
+            if (g_flag_withlock)
+            {
+                enc_spin_unlock(enclave, &g_lock);
+            }
+        }
+
+        if (g_flag_withlock)
+        {
+            enc_spin_unlock(enclave, &g_lock);
+        }
+    }
+
+    return NULL;
+}
+
+void* errno_thread2(void* para)
+{
+    int i = 0;
+    long res = 0;
+    int t_errno = 0;
+    oe_enclave_t* enclave = (oe_enclave_t*)para;
+
+    for (i = 0; i < ERRNO_TEST_TIMES; i++)
+    {
+        if (g_flag_withlock)
+        {
+            enc_spin_lock(enclave, &g_lock);
+        }
+
+        char num_str[] = "123456789987654321123456789987654321";
+        enc_strtol(enclave, &res, num_str, NULL, 10, &t_errno);
+
+        if (ERANGE != t_errno)
+        {
+            printf("[errno_thread2] expected %d, errno %d.\n", ERANGE, t_errno);
+            g_error_num |= 0x40;
+            if (g_flag_withlock)
```

Comment:
I think this unlock is not needed since there is an unlock immediately after this if statement block.

Reply:
@radhikaj Github sometimes makes this hard to review. If you expand the code, this unlock is within a loop and is therefore needed so as to match the lock.

Reply:
Actually, I was the one that misread the code :D 
Yes, upon scrolling and rereading, this unlock is not needed.